src/coreclr/nativeaot/Runtime/amd64/AllocFast.asm

```diff
@@ -65,25 +65,15 @@ NESTED_ENTRY RhpNewObject, _TEXT
         ;; Preserve the EEType in RSI
         mov         rsi, rcx
 
-        mov         r8d, [rsi + OFFSETOF__EEType__m_uBaseSize]      ; cbSize
+        xor         r8d, r8d        ; numElements
 
         ;; Call the rest of the allocation helper.
-        ;; void* RhpGcAlloc(EEType *pEEType, uint32_t uFlags, uintptr_t cbSize, void * pTransitionFrame)
+        ;; void* RhpGcAlloc(EEType *pEEType, uint32_t uFlags, uintptr_t numElements, void * pTransitionFrame)
         call        RhpGcAlloc
 
         ;; Set the new object's EEType pointer on success.
         test        rax, rax
         jz          NewOutOfMemory
-        mov         [rax + OFFSETOF__Object__m_pEEType], rsi
-
-        ;; If the object is bigger than RH_LARGE_OBJECT_SIZE, we must publish it to the BGC
-        mov         edx, [rsi + OFFSETOF__EEType__m_uBaseSize]
-        cmp         rdx, RH_LARGE_OBJECT_SIZE
```

Comment:
The large object check should be unnecessary in the fast path. There is an assert to protect it the slow path.

Removing this useless check should make the regular object allocations a tiny bit faster.